google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/flexible/AppEngineFlexibleSupportProvider.java

```diff
@@ -87,24 +110,81 @@ public boolean isSupportAlreadyAdded(@NotNull Module module,
   /** Initializes the Flexible facet by settings the default paths for app.yaml and Dockerfile
    * and generating the necessary run configurations.
    */
-  public void setupFacet(@NotNull AppEngineFlexibleFacet facet,
-      @NotNull ModifiableRootModel rootModel) {
+  public static void addSupport(@NotNull AppEngineFlexibleFacet facet,
+      @NotNull ModifiableRootModel rootModel,
+      boolean generateConfigFiles) {
     // Allows suggesting app.yaml and Dockerfile locations in facet and deployment UIs.
     VirtualFile[] contentRoots = rootModel.getContentRoots();
     AppEngineProjectService appEngineProjectService = AppEngineProjectService.getInstance();
     if (contentRoots.length > 0) {
-      facet.getConfiguration().setAppYamlPath(
+      Path appYamlPath = Paths.get(
           appEngineProjectService.getDefaultAppYamlPath(contentRoots[0].getPath()));
-      facet.getConfiguration().setDockerfilePath(
+      Path dockerfilePath = Paths.get(
           appEngineProjectService.getDefaultDockerfilePath(contentRoots[0].getPath()));
+
+      facet.getConfiguration().setAppYamlPath(appYamlPath.toString());
+      facet.getConfiguration().setDockerfilePath(dockerfilePath.toString());
+
+      // Configuration file generation.
+      Project project = facet.getModule().getProject();
+      Optional<Path> defaultAppYaml =
+          new CloudSdkAppEngineHelper(project)
+              .defaultAppYaml(FlexibleRuntime.java);
+
+      if (generateConfigFiles) {
+        if (Files.exists(appYamlPath)) {
+          int override = Messages.showYesNoDialog(project,
+              GctBundle.message("appengine.support.appyaml.existing"),
+              GctBundle.message("appengine.support.appyaml.existing.title"),
+              GoogleCloudToolsIcons.APP_ENGINE
+          );
+
+          if (override == Messages.YES) {
+            defaultAppYaml.ifPresent(
+                appYaml -> WriteCommandAction.runWriteCommandAction(project,
+                    () -> {
+                      VirtualFile appYamlVirtualFile = contentRoots[0]
+                          .findFileByRelativePath("/src/main/appengine/app.yaml");
+                      if (appYamlVirtualFile != null) {
+                        PsiFile appYamlPsiFile =
+                            PsiManager.getInstance(project).findFile(appYamlVirtualFile);
+                        if (appYamlPsiFile != null) {
+                          Document appYamlDocument =
+                              PsiDocumentManager.getInstance(project)
+                                  .getDocument(appYamlPsiFile);
+                          if (appYamlDocument != null) {
+                            try {
+                              appYamlDocument.setText(
```

Comment:
Have you tried using the combination of FileUtil.createIfDoesntExist and FileUtil.copyContent? I'm wondering if you did this then you can unify these logic paths (the case was the file file already exists, and the case were it does not) and simplify this method a lot. Also, this _might_ sidestep the line separator issue.  As is, I find this method a little too messy.